File: Arrays/isolerElements.js
/**
 * Isoler les éléments doublons d'un tableau dans un autre, sans changer l'ordre des éléments.
 * @param {Array} arr - Le tableau d'origine.
 * @param {Array} sub - Le tableau dans lequel seront stockés les éléments isolés.
 * @param {Object} e - L'élément à isoler.
 * @returns {Array} Le tableau d'origine moins les éléments isolés, et le tableau d'éléments isolés.
 */
const isolerElements = (arr, sub, e) => {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === e) {
      sub.push(arr[i]);
      arr.splice(i, 1);
      i--;
    }
  }
  return [arr, sub];
};

// Test
console.log(
  isolerElements([0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1], [], 1)
);

/*
[
  [
    0, 0, 0, 0,
    0, 0, 0, 0
  ],
  [
    1, 1, 1, 1,
    1, 1, 1, 1
  ]
]
*/

File: Python/Arrays/isolerElements.py
def isolerElements(arr: list, sub: list, e: int) -> list:
    """
    Description:
        Isoler les éléments doublons d'un tableau dans un autre, sans changer l'ordre des éléments.
    
    Arguments:
        arr: {list} -- Tableau d'entiers.
        sub: {list} -- Tableau d'entiers.
        e: {int} -- Entier.

    Return:
        {list} -- Tableau d'entiers.
    
    Exemple:
        >>> isolerElements([0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1], [], 1)
        [[0, 0, 0, 0, 0, 0, 0, 0], [1, 1, 1, 1, 1, 1, 1, 1]]
    """
    for i in range(len(arr)):
        if arr[i] == e:
            sub.append(arr[i])
            arr[i] = None
    while None in arr:
        arr.remove(None)
    return [arr, sub]